python/nav/mibs/pwt_3phase_mibv1.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2008-2011 Uninett AS
+#
+# This file is part of Network Administration Visualized (NAV).
+#
+# NAV is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License version 2 as published by
+# the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.  You should have received a copy of the GNU General Public
+# License along with NAV. If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+A class that tries to retrieve all sensors from Powertek PDU.
+
+Uses the vendor-specifica PWT_3Phase_MIBv1_v2.10-MIB to detect and collect
+sensor-information.
+
+Please note!
+-- not a complete implementation of the mib --
+
+"""
+from django.utils import six
+from django.utils.six import itervalues
+from twisted.internet import defer
+
+from nav.mibs import reduce_index
+from nav.mibs import mibretriever
+from nav.models.manage import Sensor
+from nav.oids import OID
+
+# from .itw_mib import for_table
+
+def for_table(table_name):
+    """Used for annotating functions to process the returned
+    tables"""
+    if not hasattr(for_table, 'map'):
+        for_table.map = {}
+
+    def decorate(method):
+        """Setup link between table and function"""
+        name = method.func_name if six.PY2 else method.__name__
+        for_table.map[table_name] = name
+        return method
+
+    return decorate
+
+class Pwt3PhaseV1Mib(mibretriever.MibRetriever):
+    """A class that tries to retrieve all sensors from Powertek PDU"""
+    from nav.smidumps.pwt_3phasev2_10_mibv1 import MIB as mib
+
+    oid_name_map = dict((OID(attrs['oid']), name)
+                        for name, attrs in mib['nodes'].items())
+
+    lowercase_nodes = dict((key.lower(), key)
+                           for key in mib['nodes'])
+
+    def _debug(self, msg, *args, **kwargs):
```

Comment:
These two internal logging methods seem entirely superfluous to me. The class name is already part of the logger name - this would only output the class name twice, would it not?

Reply:
Yes, you are right :)